drivers/power/adp1050/iio_adp1050.c

```diff
@@ -0,0 +1,1305 @@
+		.show = adp1050_iio_read_status,
+		.priv = ADP1050_STATUS_CML_TYPE
+	},
+	{
+		.name = "status_word",
+		.show = adp1050_iio_read_status,
+		.priv = ADP1050_STATUS_WORD_TYPE
+	},
+	END_ATTRIBUTES_ARRAY
+};
+
+static struct iio_channel adp1050_channels[] = {
+	{
+		.name = "vin",
+		.ch_type = IIO_VOLTAGE,
+		.indexed = 1,
+		.channel = ADP1050_IIO_VIN_CHAN,
+		.address = ADP1050_IIO_VIN_CHAN,
+		.attributes = adp1050_input_attrs,
+		.ch_out = false
+	},
+	{
+		.name = "iin",
+		.ch_type = IIO_CURRENT,
+		.indexed = 1,
+		.channel = ADP1050_IIO_IIN_CHAN,
+		.address = ADP1050_IIO_IIN_CHAN,
+		.attributes = adp1050_input_attrs,
+		.ch_out = false,
+	},
+	{
+		.name = "vout",
+		.ch_type = IIO_VOLTAGE,
+		.indexed = 1,
+		.channel = ADP1050_IIO_VOUT_CHAN,
+		.address = ADP1050_IIO_VOUT_CHAN,
+		.attributes = adp1050_input_attrs,
+		.ch_out = false,
+	},
+	{
+		.name = "temperature",
+		.ch_type = IIO_TEMP,
+		.indexed = 1,
+		.channel = ADP1050_IIO_TEMP_CHAN,
+		.address = ADP1050_IIO_TEMP_CHAN,
+		.attributes = adp1050_input_attrs,
+		.ch_out = false,
+	},
+	{
+		.name = "outa",
+		.ch_type = IIO_VOLTAGE,
+		.indexed = 1,
+		.channel = ADP1050_IIO_OUTA_CHAN,
+		.address = ADP1050_IIO_OUTA_CHAN,
+		.attributes = adp1050_output_attrs,
+		.ch_out = true,
+	},
+	{
+		.name = "outb",
+		.ch_type = IIO_VOLTAGE,
+		.indexed = 1,
+		.channel = ADP1050_IIO_OUTB_CHAN,
+		.address = ADP1050_IIO_OUTB_CHAN,
+		.attributes = adp1050_output_attrs,
+		.ch_out = true,
+	},
+	{
+		.name = "sr1",
+		.ch_type = IIO_VOLTAGE,
+		.indexed = 1,
+		.channel = ADP1050_IIO_SR1_CHAN,
+		.address = ADP1050_IIO_SR1_CHAN,
+		.attributes = adp1050_output_attrs,
+		.ch_out = true,
+	},
+	{
+		.name = "sr2",
+		.ch_type = IIO_VOLTAGE,
+		.indexed = 1,
+		.channel = ADP1050_IIO_SR2_CHAN,
+		.address = ADP1050_IIO_SR2_CHAN,
+		.attributes = adp1050_output_attrs,
+		.ch_out = true,
+	}
+};
+
+static struct iio_device adp1050_iio_dev = {
+	.num_ch = NO_OS_ARRAY_SIZE(adp1050_channels),
+	.channels = adp1050_channels,
+	.attributes = adp1050_global_attrs,
+	.debug_attributes = adp1050_debug_attrs
+};
+
+/**
+ * @brief Handles the read request for raw attribute.
+ * @param dev     - The iio device structure.
+ * @param buf	  - Command buffer to be filled with requested data.
+ * @param len     - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv    - Command attribute id.
+ * @return ret    - Result of the reading procedure.
+ * 		    In case of success, the size of the read data is returned.
+*/
+static int adp1050_iio_read_raw(void *dev, char *buf, uint32_t len,
+				const struct iio_ch_info *channel,
+				intptr_t priv)
+{
+	int ret;
+	uint16_t mant, val;
+	uint8_t exp;
+	struct adp1050_iio_desc *iio_adp1050 = dev;
+	struct adp1050_desc *adp1050 = iio_adp1050->adp1050_desc;
+
+	switch (channel->address) {
+	case ADP1050_IIO_VIN_CHAN:
+		ret = adp1050_read_value(adp1050, &mant, &exp, ADP1050_VIN);
+		if (ret)
+			return ret;
```

Comment:
can remove all these checks from the switch  and add the if condition a single time after the switch.
see all functions where applicable